### ISSUE TITLE
SNOW-1038158: Python Connector Diagnostic fails because system$whitelist is deprecated

### ISSUE THREAD

**Issue by sfc-gh-cconner** [snowflakedb/snowflake-connector-python#1870]
### Python version

3.10.12

### Operating system and processor architecture

macOS-14.3-arm64-arm-64bit

### Installed packages

```python
asn1crypto==1.5.1
build==1.0.3
certifi==2023.7.22
cffi==1.15.1
charset-normalizer==3.2.0
coverage==7.4.1
cryptography==41.0.3
Cython==3.0.8
exceptiongroup==1.2.0
execnet==2.0.2
filelock==3.12.4
idna==3.4
iniconfig==2.0.0
more-itertools==10.2.0
numpy==1.26.3
oscrypto==1.3.0
packaging==23.1
pendulum==3.0.0
pexpect==4.9.0
platformdirs==3.8.1
pluggy==1.4.0
ptyprocess==0.7.0
pycparser==2.21
pycryptodomex==3.18.0
PyJWT==2.8.0
pyOpenSSL==23.2.0
pyproject_hooks==1.0.0
pytest==7.4.4
pytest-cov==4.1.0
pytest-rerunfailures==13.0
pytest-timeout==2.2.0
pytest-xdist==3.5.0
python-dateutil==2.8.2
pytz==2023.3.post1
pytzdata==2020.1
requests==2.31.0
six==1.16.0
snowflake-connector-python @ file:///Users/cconner/snowflake/source/forks/snowflake-connector-python/dist/snowflake_connector_python-3.7.0-cp310-cp310-macosx_13_0_arm64.whl#sha256=4c12190e7bc2312ec49bbf0d054b475b15e2a081796ce8edc46b5f5063deb6a4
sortedcontainers==2.4.0
time-machine==2.13.0
tomli==2.0.1
tomlkit==0.12.1
typing_extensions==4.7.1
tzdata==2023.4
urllib3==1.26.16
```

### What did you do?

```python
Tried to run the python connector diagnostic and it failed because the result of the system$whitelist is now:

This function has been deprecated. Use SYSTEM$ALLOWLIST instead.
```

### What did you expect to see?

I expected to see the diagnostic report.

### Can you set logging to DEBUG and collect the logs?

```bash
N/A
```

**Comment by sfc-gh-dszmolka** [comment]
thanks for raising this issue ;) and the https://github.com/snowflakedb/snowflake-connector-python/pull/1871 too !

**Comment by sfc-gh-dszmolka** [comment]
PR merged, waiting for release

**Comment by sfc-gh-dszmolka** [comment]
fix released with v3.8.0
